workflow/rules/quant.smk

```diff
@@ -1,28 +1,336 @@
 rule kallisto_index:
     input:
-        "resources/transcriptome.cdna.fasta",
+        fasta="resources/transcriptome_clean.{type}.fasta"
+        if is_3prime_experiment
+        else "resources/transcriptome.cdna.fasta",
     output:
-        "results/kallisto/transcripts.idx",
+        index="results/kallisto_{type}/transcripts.{type}.idx",
     log:
-        "results/logs/kallisto/index.log",
-    conda:
-        "../envs/kallisto.yaml"
-    shell:
-        "kallisto index -i {output} {input} 2> {log}"
+        "results/logs/kallisto_{type}/index.{type}.log",
+    wildcard_constraints:
+        type="cdna|3prime",
+    threads: 1
+    wrapper:
+        "v1.17.4/bio/kallisto/index"
 
 
 rule kallisto_quant:
     input:
-        fq=get_trimmed,
-        idx="results/kallisto/transcripts.idx",
+        fastq=kallisto_quant_input,
+        index="results/kallisto_{type}/transcripts.{type}.idx",
     output:
-        directory("results/kallisto/{sample}-{unit}"),
+        kallisto_folder=directory("results/kallisto_{type}/{sample}-{unit}"),
     log:
-        "results/logs/kallisto/quant/{sample}-{unit}.log",
+        "results/logs/kallisto_{type}/quant/{sample}-{unit}.log",
     params:
         extra=kallisto_params,
+    threads: 5
+    wildcard_constraints:
+        type="cdna|3prime",
+    wrapper:
+        "v1.17.4/bio/kallisto/quant"
+
+
```

Comment:
Move the 3prime specific quant rules below into a separate file quant_3prime.smk

Reply:
Noted. Changed